File: topics/02-key-value-store/java/src/main/java/dev/reinvent/wheel/t02/KeyValueStoreApp.java
package dev.reinvent.wheel.t02;

public final class KeyValueStoreApp {
    public record CliArgs(String input) {
    }

    private KeyValueStoreApp() {
    }

    public static CliArgs parseArgs(String[] args) {
        if (args.length < 1 || args.length > 1) {
            throw new IllegalArgumentException("usage: t02-key-value-store-java <input>");
        }
        String input = args[0];
        return new CliArgs(input);
    }

    public static String helloMessage(CliArgs args) {
        var ignored = args;
        return "hello kv-store\n";
    }

    public static void main(String[] args) {
        System.out.print(helloMessage(parseArgs(args)));
    }
}

File: topics/02-key-value-store/kotlin/src/main/kotlin/dev/reinvent/wheel/t02/KeyValueStoreApp.kt
package dev.reinvent.wheel.t02

object KeyValueStoreApp {
    data class CliArgs(val input: String)

    fun parseArgs(args: Array<String>): CliArgs {
        if (args.size < 1 || args.size > 1) {
            throw IllegalArgumentException("usage: t02-key-value-store-kotlin <input>")
        }
        val input = args[0]
        return CliArgs(input)
    }

    fun helloMessage(args: CliArgs): String {
        val ignored = args
        return "hello kv-store\n"
    }
}

fun main(args: Array<String>) {
    print(KeyValueStoreApp.helloMessage(KeyValueStoreApp.parseArgs(args)))
}
